docs/pages/versions/unversioned/sdk/device.md

```diff
@@ -0,0 +1,273 @@
+
+  Gets the device brand.
+
+  - iOS: `"Apple"`
+  - Android: eg. `"Xiaomi"`
+
+- `Device.carrier: string`
+
+  Gets the carrier name (network operator).
+
+- `Device.manufacturer: string`
+
+  Gets the device manufacturer.
+
+  - iOS: `"Apple"`
+  - Android: eg. `"Google"`
+
+- `Device.model: string`
+
+  **iOS warning**: The list with device names is maintained by the community and could lag new devices. It is recommended to use `deviceId` since it's more reliable and always up-to-date with new iOS devices. We do accept pull requests that add new iOS devices to the list with device names.
+
+  Gets the device model.
+
+  - iOS: eg. `"iPhone XS Max"`
+  - Android: eg. `"Pixel 2"`
+
+- `Device.systemName: string`
+
+  Gets the device OS name.
+
+- `Device.deviceId: string`
+
+  Gets the device ID.
+
+- `Device.totalMemory: number`
+
+  Gets the device total memory, in bytes.
+
+- `Device.uniqueId: string`
+
+  Gets the device unique ID.
+
+- `Device.isTablet: boolean`
+
+  Tells if the device is a tablet.
+
+- `Device.deviceType: string`
+
+  Returns the device's type as a string, which will be one of:
+
+  - `Handset`
+  - `Tablet`
+  - `Tv`
+  - `Unknown`
+
+- `Device.supportedABIs: string[]`
+
+  Returns a list of supported processor architecture version
+
+  **Examples**
+
+  ```js
+  Device.supportedABIs; // [ "arm64 v8", "Intel x86-64h Haswell", "arm64-v8a", "armeabi-v7a", "armeabi" ]
+  ```
+
+### Methods
+
+- `Device.hasNotch()`
+- `Device.hasSystemFeatureAsync(feature)` (Android only)
+- `Device.getIpAddressAsync()`
+- `Device.getMACAddressAsync()`
+- `Device.isAirplaneModeEnabledAsync()` (Android only)
+- `Device.isPinOrFingerprintSet()`
+- `Device.getFreeDiskStorageAsync()`
+- `Device.getUserAgentAsync()`
+- `Device.getTotalDiskCapacityAsync()`
+- `Device.getSerialNumberAsync()` (Android only)
+
+## Methods
+
+### `Device.hasNotch()`
+
+Tells if the device has a notch.
+
+#### Returns
+
+A boolean that represents the support for notch display.
+
+**Examples**
+
+```js
+import { hasNotch } from 'expo-device';
+
+hasNotch(); // true or false
+```
+
+### `Device.getUserAgentAsync()`
+
+Gets the device User Agent.
+
+#### Returns
+
+A Promise that resolves to a string of User Agent.
+
+**Examples**
+
+```js
+Device.getUserAgentAsync().then(userAgent => {
+  //Dalvik/2.1.0 (Linux; U; Android 9; Pixel 2 Build/PQ3A.190505.001)
+});
+```
+
+### `Device.getFreeDiskStorageAsync()`
+
+Gets available storage size, in bytes.
+
+#### Returns
+
+A promise of number that represents the storage size.
+
+**Examples**
+
+```js
+Device.getFreeDiskStorageAsync().then(storage => {
+  //5608296448
+});
+```
+
+### `Device.getTotalDiskCapacityAsync()`
+
+Gets full disk storage size, in bytes.
+
+#### Returns
+
+A promise of number that represents the full disk storage size.
+
+**Examples**
+
+```js
+Device.getTotalDiskCapacityAsync().then(storage => {
+  //17179869184
+});
+```
+
+### `Device.getIpAddressAsync()`
+
+Gets the device current IP address.
```

Comment:
Also seems like it'd be more apropos to put the IP- & MAC-address methods into a networking module. Perhaps the airplane-mode check belongs there too. (@brentvatne what do you think about creating a derivative of r-n-netinfo like `expo-network-info` so that (a) we don't need to keep up to date with netinfo and also can put the IP/MAC-address methods in there?)